src/tool/hpcrun/sample-sources/level0.c

```diff
@@ -195,8 +196,11 @@ METHOD_FN(finalize_event_list)
     monitor_real_exit(-1);
   }
 #endif
-
   level0_init();
+
+  // Init records
+  gpu_trace_init();
```

Comment:
shouldn't this only be done if tracing is enabled? is that handled inside the gpu tracing code?

Reply:
After a further look, I am not sure... 

`gpu_trace_init` is also always called for nvidia code:

https://github.com/HPCToolkit/hpctoolkit/blob/develop/src/tool/hpcrun/sample-sources/nvidia.c#L272

And the implementation of `gpu_trace_init` does not seem to examine the command line argument for tracing.

Reply:
I read the code. It simply initializes variables used to coordinate any tracing that may occur. Performing this initialization is harmless. I approve.